### ISSUE TITLE
Stadia controller axis-related input events not coherent

### ISSUE THREAD

**Issue by PKGaspi** [godotengine/godot#36370]
**Godot version:**
3.2 stable

**OS/device including version:**
Arch Linux 5.5.4

**Issue description:**
The Stadia controller behaves very different to other controllers. These are the anomalies that I found:

- Joystick events don't trigger `_input(event)` by normal use (not tested with `_gui_input(event)` nor `_unhandeled_input(event)`). Instead, they get triggered when the Joystick is clicked, then moved and finally released with some junk values such as `0` or `nan`. Using `Input.get_joy_axis(device, axis)` doesn't read any values with any joystick axis (0, 1, 2 and 3).
- Triggers (L2 and R2) call the `_input(event)` event when pressed, but again with a different index than expected. It is index 5 for L2 and 4 for R2 but it is represented in the mapping configuration as 6 and 7 respectively.

**This is just a hypothesis, but the cause of the whole issue might just be weird axis assignation.** 

I don't know if this issue is present in other controllers. I know that Stadia support is not planed for now, but controller support is a different topic.

**Steps to reproduce:**
1.- Play the reproduction project.
2.- Compare outputs between the Stadia controller and a different one.

**Minimal reproduction project:**
[controller_inputs.zip](https://github.com/godotengine/godot/files/4227231/controller_inputs.zip)

By the way, testing for this issue I noticed that both joystick up and joystick down actions get triggered at the same time when moving the joystick only up or only down. The same goes for both joysticks and left/right. It happens on the xbox controller as well. I don't know if this is considered to work as intended or if I should open a different issue for it.

**Comment by jonas-koeritz** [comment]
I've tried this and I can verify that everything is working as expected except for the Left and Right Analog sticks. All other Buttons (including R3 and L3) are working correctly, the triggers work correctly and return usable values but the sticks don't seem to do anything.
As other applications don't have any problem to detect the axis movement (for example [The HTML5 Gamepad Tester](https://html5gamepad.com/) and correctly assign them to axis 0 through 3 there should be a bug in Godot.

**Comment by jonas-koeritz** [comment]
The issue might be that the game controller DB is not up to date. There is only a mapping for windows available in Godot. https://github.com/gabomdq/SDL_GameControllerDB/blob/master/gamecontrollerdb.txt contains two more mappings for Linux and macOS.

**Comment by Calinou** [comment]
@jonas-koeritz See also #8368.

**Comment by fcole90** [comment]
I'm using version 3.3.1.stable.mono and I just tested with the minimal project.

If I connect the controller while the "game" is running I obtain a seemingly random list of actions:
```
--- Debugging process started ---
Godot Engine v3.3.1.stable.mono.official - https://godotengine.org
OpenGL ES 3.0 Renderer: GeForce GTX 1650/PCIe/SSE2
OpenGL ES Batching: ON
 
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
--------------------[...]
 [output overflow, print less text!]
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
--------------------[...]
 [output overflow, print less text!]
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
--------------------[...]
 [output overflow, print less text!]
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
--------------------[...]
 [output overflow, print less text!]
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	a[...]
 [output overflow, print less text!]
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
---------------------------
right_joystick_left
right_joystick_right
device	axis	axis_value
0	2	-nan
---------------------------
right_joystick_down
right_joystick_up
device	axis	axis_value
0	3	-nan
---------------------------
left_joystick_left
left_joystick_right
device	axis	axis_value
0	0	-nan
---------------------------
left_joystick_down
left_joystick_up
device	axis	axis_value
0	1	-nan
--------------------[...]
 [output overflow, print less text!]
--- Debugging process stopped ---
```

If I have the Stadia controller connected in advance, only L2 and R2 are detected correctly, as analogic triggers. The output of pressing first R2 and then L2 is the following:

```
R2
device	axis	axis_value
0	7	0.007843
---------------------------
R2
device	axis	axis_value
0	7	0.035294
---------------------------
R2
device	axis	axis_value
0	7	0.062745
---------------------------
R2
device	axis	axis_value
0	7	0.082353
---------------------------
R2
device	axis	axis_value
0	7	0.105882
---------------------------
R2
device	axis	axis_value
0	7	0.133333
---------------------------
R2
device	axis	axis_value
0	7	0.156863
---------------------------
R2
device	axis	axis_value
0	7	0.176471
---------------------------
R2
device	axis	axis_value
0	7	0.2
---------------------------
R2
device	axis	axis_value
0	7	0.219608
---------------------------
R2
device	axis	axis_value
0	7	0.243137
---------------------------
R2
device	axis	axis_value
0	7	0.254902
---------------------------
R2
device	axis	axis_value
0	7	0.278431
---------------------------
R2
device	axis	axis_value
0	7	0.305882
---------------------------
R2
device	axis	axis_value
0	7	0.321569
---------------------------
R2
device	axis	axis_value
0	7	0.345098
---------------------------
R2
device	axis	axis_value
0	7	0.364706
---------------------------
R2
device	axis	axis_value
0	7	0.380392
---------------------------
R2
device	axis	axis_value
0	7	0.407843
---------------------------
R2
device	axis	axis_value
0	7	0.447059
---------------------------
R2
device	axis	axis_value
0	7	0.494118
---------------------------
R2
device	axis	axis_value
0	7	0.541176
---------------------------
R2
device	axis	axis_value
0	7	0.588235
---------------------------
R2
device	axis	axis_value
0	7	0.627451
---------------------------
R2
device	axis	axis_value
0	7	0.662745
---------------------------
R2
device	axis	axis_value
0	7	0.694118
---------------------------
R2
device	axis	axis_value
0	7	0.713726
---------------------------
R2
device	axis	axis_value
0	7	0.733333
---------------------------
R2
device	axis	axis_value
0	7	0.74902
---------------------------
R2
device	axis	axis_value
0	7	0.776471
---------------------------
R2
device	axis	axis_value
0	7	1
---------------------------
R2
device	axis	axis_value
0	7	0.996078
---------------------------
R2
device	axis	axis_value
0	7	0.854902
---------[...]
 [output overflow, print less text!]
L2
device	axis	axis_value
0	6	0.019608
---------------------------
L2
device	axis	axis_value
0	6	0.039216
---------------------------
L2
device	axis	axis_value
0	6	0.05098
---------------------------
L2
device	axis	axis_value
0	6	0.066667
---------------------------
L2
device	axis	axis_value
0	6	0.090196
---------------------------
L2
device	axis	axis_value
0	6	0.101961
---------------------------
L2
device	axis	axis_value
0	6	0.117647
---------------------------
L2
device	axis	axis_value
0	6	0.133333
---------------------------
L2
device	axis	axis_value
0	6	0.14902
---------------------------
L2
device	axis	axis_value
0	6	0.180392
---------------------------
L2
device	axis	axis_value
0	6	0.219608
---------------------------
L2
device	axis	axis_value
0	6	0.270588
---------------------------
L2
device	axis	axis_value
0	6	0.333333
---------------------------
L2
device	axis	axis_value
0	6	0.403922
---------------------------
L2
device	axis	axis_value
0	6	0.47451
---------------------------
L2
device	axis	axis_value
0	6	0.541176
---------------------------
L2
device	axis	axis_value
0	6	0.611765
---------------------------
L2
device	axis	axis_value
0	6	0.682353
---------------------------
L2
device	axis	axis_value
0	6	0.756863
---------------------------
L2
device	axis	axis_value
0	6	0.831373
---------------------------
L2
device	axis	axis_value
0	6	0.905882
---------------------------
L2
device	axis	axis_value
0	6	0.980392
---------------------------
L2
device	axis	axis_value
0	6	1
---------------------------
L2
device	axis	axis_value
0	6	0.984314
---------------------------
L2
device	axis	axis_value
0	6	0.823529
---------------------------
L2
device	axis	axis_value
0	6	0.658824
---------------------------
L2
device	axis	axis_value
0	6	0.494118
---------------------------
L2
device	axis	axis_value
0	6	0.341176
---------------------------
L2
device	axis	axis_value
0	6	0.196078
---------------------------
L2
device	axis	axis_value
0	6	0.05098
---------------------------
L2
device	axis	axis_value
0	6	0
---------------------------
R2
device	axis	axis_value
0	7	0.160784
---------------------------
R2
device	axis	axis_value
0	7	0.313726
---------------------------
R2
device	axis	axis_value
0	7	0.443137
---------------------------
R2
device	axis	axis_value
0	7	0.564706
---------------------------
R2
device	axis	axis_value
0	7	0.67451
---------------------------
R2
device	axis	axis_value
0	7	0.760784
---------------------------
R2
device	axis	axis_value
0	7	0.831373
---------------------------
R2
device	axis	axis_value
0	7	0.894118
---------------------------
R2
device	axis	axis_value
0	7	0.952941
---------------------------
R2
device	axis	axis_value
0	7	1
---------------------------
R2
device	axis	axis_value
0	7	0.976471
---------------------------
R2
device	axis	axis_value
0	7	0.858824
---------------------------
R2
device	axis	axis_value
0	7	0.737255
---------------------------
R2
device	axis	axis_value
0	7	0.6
---------------------------
R2
device	axis	axis_value
0	7	0.458824
---------------------------
R2
device	axis	axis_value
0	7	0.313726
---------------------------
R2
device	axis	axis_value
0	7	0.247059
---------------------------
R2
device	axis	axis_value
0	7	0.101961
---------------------------
R2
device	axis	axis_value
0	7	0
---------------------------
L2
device	axis	axis_value
0	6	0.007843
---------------------------
L2
device	axis	axis_value
0	6	0.094118
---------------------------
L2
device	axis	axis_value
0	6	0.168627
---------------------------
L2
device	axis	axis_value
0	6	0.247059
---------------------------
L2
device	axis	axis_value
0	6	0.321569
---------------------------
L2
device	axis	axis_value
0	6	0.388235
---------------------------
L2
device	axis	axis_value
0	6	0.454902
---------------------------
L2
device	axis	axis_value
0	6	0.529412
---------------------------
L2
device	axis	axis_value
0	6	0.611765
---------------------------
L2
device	axis	axis_value
0	6	0.701961
---------------------------
L2
device	axis	axis_value
0	6	0.780392
---------------------------
L2
device	axis	axis_value
0	6	0.854902
---------------------------
L2
device	axis	axis_value
0	6	0.937255
---------------------------
L2
device	axis	axis_value
0	6	1
---------------------------
L2
device	axis	axis_value
0	6	0.980392
---------------------------
L2
device	axis	axis_value
0	6	0.811765
---------------------------
L2
device	axis	axis_value
0	6	0.623529
---------------------------
L2
device	axis	axis_value
0	6	0.517647
---------------------------
L2
device	axis	axis_value
0	6	0.415686
---------------------------
L2
device	axis	axis_value
0	6	0.305882
---------------------------
L2
device	axis	axis_value
0	6	0.203922
---------------------------
L2
device	axis	axis_value
0	6	0.101961
---------------------------
L2
device	axis	axis_value
0	6	0
---------------------------
```

**Comment by Powerbyte7** [comment]
Looks like this issue has been fixed in master, probably with f41c72c5384ff9ce34f60380d4f27b9af0904298. Here's a small test I did:

[Video](https://user-images.githubusercontent.com/40431794/154735483-c7aaf47e-e285-4536-ba78-3dfa7a29e3a5.mp4)
[Project file](https://github.com/godotengine/godot/files/8099311/input_test.zip)

I'd love to see this fix back-ported to 3.x though, if that's possible.

**Comment by fcole90** [comment]
Hi, any update on this? I'd like to try backporting this if someone helps me getting started 😊

**Comment by fcole90** [comment]
Hi, I just recompiled from source, from branch `3.x`, and I cannot reproduce the issue in Ubuntu 22.04. Can someone double-check?

```
Godot Engine v3.5.beta.custom_build.fa9b77dbe - https://godotengine.org
OpenGL ES 2.0 Renderer: Mesa Intel(R) UHD Graphics 630 (CFL GT2)
```

My steps:

Install deps:
```sh
sudo apt install build-essential scons pkg-config libx11-dev libxcursor-dev libxinerama-dev     libgl1-mesa-dev libglu-dev libasound2-dev libpulse-dev libudev-dev libxi-dev libxrandr-dev yasm
```

```
git clone https://github.com/godotengine/godot.git
cd godot
git checkout 3.x
scons -j$(nproc) platform=x11
```

then run the bin: `./bin/godot.x11.tools.64`

**Comment by nathanfranke** [comment]
Can confirm this is no longer reproducible in 21c304839.

**Comment by akien-mga** [comment]
This might have been fixed by #56239 (CC @madmiraal)
